data-prepper-core/src/main/java/org/opensearch/dataprepper/core/sourcecoordination/LeaseBasedSourceCoordinator.java

```diff
@@ -343,14 +343,37 @@ private void updatePartitionOwnership(final String partitionKey, final Duration
 
     @Override
     public void giveUpPartition(final String partitionKey) {
-        giveUpPartitionInternal(partitionKey, null);
+        giveUpPartitionInternalWithRetries(partitionKey, null);
     }
 
     @Override
     public void giveUpPartition(final String partitionKey, final Instant priorityTimestamp) {
-        giveUpPartitionInternal(partitionKey, priorityTimestamp);
+        giveUpPartitionInternalWithRetries(partitionKey, priorityTimestamp);
     }
 
+    public void giveUpPartitionInternalWithRetries(final String partitionKey, final Instant priorityTimestamp) {
```

Comment:
May be lower blast radius to add a new function to the interface


```
    @Override
    public void giveUpPartition(final String partitionKey, final Instant priorityTimestamp, final int maxRetries) {
        giveUpPartitionInternalWithRetries(partitionKey, priorityTimestamp);
    }
```

Then just use this method in S3 scan folder partition for now. Also the internal method should be private.